proxy/src/serverless/sql_over_http.rs

```diff
@@ -171,7 +172,8 @@ fn get_conn_info(
         .and_then(|h| h.to_str().ok())
         .and_then(|h| h.split(':').next());
 
-    if hostname != sni_hostname {
+    // sni_hostname has to be either the same as hostname or the one used in serverless driver.
+    if sni_hostname != SERVERLESS_DRIVER_SNI_HOSTNAME || sni_hostname != hostname {
         return Err(anyhow::anyhow!("mismatched SNI hostname and hostname"));
     } else if let Some(h) = host_header {
         if h != hostname {
```

Comment:
same needs to happen here, afaiu

Reply:
ah, and you need to make this an `&&`, not an or 😅

Reply:
Oh, thanks!

Reply:
the host header validation still needs to be changed

Reply:
see
```
curl -k -v 'https://api.cloud.chloroform.aws.neon.build/sql' \
  -H 'Neon-Connection-String: postgresql://***@***.cloud.chloroform.aws.neon.build/neondb' \
  -H 'Content-Type: application/json' \
  -H 'Neon-Raw-Text-Output: true' \
  -H 'Neon-Array-Mode: false' \
  --data '{
    "query":"SHOW server_version_num",
    "params":[]
  }' | jq

> POST /sql HTTP/1.1
> Host: api.cloud.chloroform.aws.neon.build
> User-Agent: curl/8.1.2
> Accept: */*
> Neon-Connection-String: postgresql://***@***.cloud.chloroform.aws.neon.build/neondb
> Content-Type: application/json
> Neon-Raw-Text-Output: true
> Neon-Array-Mode: false
> Content-Length: 60
```

Reply:
Thanks a lot! Updated and added test